bluefog/common/operations.cc

```diff
@@ -765,7 +779,17 @@ void PerformOperationWithFusion(std::vector<TensorTableEntry>& entries) {
       first_entry.context,
       [&]() { timeline.ActivityStartAll(entries, "INIT_FUSION_BUFFER"); },
       [&]() { timeline.ActivityEndAll(entries); });
-  if (!status.ok()) {
+  
+  Status status_dst_weight = Status::OK();
+  if (first_entry.dst_weighting_enabled) {
+    status_dst_weight = bluefog_global.fusion_buffer.InitializeWeightBuffer(
+      bluefog_global.tensor_fusion_threshold, mpi_context.size_,
```

Comment:
I reuse the current threshold, as it works for all operations. Do you think a constant threshold only for dst_weight is necessary?

Reply:
Use give another name in bluefog_global. like bluefog_global.fusion_threshold_for_dst_weight. It is for clarification to remind us that we create two memories instead of one.

Reply:
Done.